src/app/shared/services/base.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { environment } from '../../../environments/environment';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { catchError, retry } from 'rxjs/operators';
+import { map } from 'rxjs/operators';
+import { Deserializer } from 'ts-jsonapi';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export abstract class BaseService {
+  DEFAULT_HEADER: object = {
+    'Content-Type': 'application/json'
+  };
+
+  deserializer: Deserializer;
+
+  constructor(private http: HttpClient) {
+    this.deserializer = new Deserializer({
+      keyForAttribute: 'camelCase'
+    });
+  }
+
+  post(endpoint: string, body: {}): Observable<any> {
```

Comment:
Why is `Observable<any>` needed? I want to learn 😁

Reply:
@olivierobert So when we use the services we can `.subscribe` for response and error later 🤔